### ISSUE TITLE
agent-addr option is ignored in default-server line

### ISSUE THREAD

**Issue by Xiol** [haproxy/haproxy#408]
## Output of `haproxy -vv` and `uname -a`

```
HA-Proxy version 2.0.8 2019/10/23 - https://haproxy.org/
Build options :
  TARGET  = linux-glibc
  CPU     = generic
  CC      = gcc
  CFLAGS  = -O2 -g -fno-strict-aliasing -Wdeclaration-after-statement -fwrapv -Wno-unused-label -Wno-sign-compare -Wno-unused-parameter -Wno-old-style-declaration -Wno-ignored-qualifiers -Wno-clobbered -Wno-missing-field-initializers -Wtype-limits
  OPTIONS = USE_PCRE_JIT=1 USE_REGPARM=1 USE_LINUX_TPROXY=1 USE_CRYPT_H=1 USE_GETADDRINFO=1 USE_OPENSSL=1 USE_ZLIB=1 USE_SYSTEMD=1

Feature list : +EPOLL -KQUEUE -MY_EPOLL -MY_SPLICE +NETFILTER -PCRE +PCRE_JIT -PCRE2 -PCRE2_JIT +POLL -PRIVATE_CACHE +THREAD -PTHREAD_PSHARED +REGPARM -STATIC_PCRE -STATIC_PCRE2 +TPROXY +LINUX_TPROXY +LINUX_SPLICE +LIBCRYPT +CRYPT_H -VSYSCALL +GETADDRINFO +OPENSSL -LUA +FUTEX +ACCEPT4 -MY_ACCEPT4 +ZLIB -SLZ +CPU_AFFINITY +TFO +NS +DL +RT -DEVICEATLAS -51DEGREES -WURFL +SYSTEMD -OBSOLETE_LINKER +PRCTL +THREAD_DUMP -EVPORTS

Default settings :
  bufsize = 16384, maxrewrite = 1024, maxpollevents = 200

Built with multi-threading support (MAX_THREADS=64, default=12).
Built with OpenSSL version : OpenSSL 1.1.1d  10 Sep 2019
Running on OpenSSL version : OpenSSL 1.1.1d  10 Sep 2019
OpenSSL library supports TLS extensions : yes
OpenSSL library supports SNI : yes
OpenSSL library supports : TLSv1.0 TLSv1.1 TLSv1.2 TLSv1.3
Built with network namespace support.
Built with transparent proxy support using: IP_TRANSPARENT IPV6_TRANSPARENT IP_FREEBIND
Built with zlib version : 1.2.7
Running on zlib version : 1.2.7
Compression algorithms supported : identity("identity"), deflate("deflate"), raw-deflate("deflate"), gzip("gzip")
Built with PCRE version : 8.32 2012-11-30
Running on PCRE version : 8.32 2012-11-30
PCRE library supports JIT : yes
Encrypted password support via crypt(3): yes
Built with the Prometheus exporter as a service

Available polling systems :
      epoll : pref=300,  test result OK
       poll : pref=200,  test result OK
     select : pref=150,  test result OK
Total: 3 (3 usable), will use epoll.

Available multiplexer protocols :
(protocols marked as <default> cannot be specified using 'proto' keyword)
              h2 : mode=HTX        side=FE|BE     mux=H2
              h2 : mode=HTTP       side=FE        mux=H2
       <default> : mode=HTX        side=FE|BE     mux=H1
       <default> : mode=TCP|HTTP   side=FE|BE     mux=PASS

Available services :
	prometheus-exporter

Available filters :
	[SPOE] spoe
	[COMP] compression
	[CACHE] cache
	[TRACE] trace

-----
Linux <redacted> 3.10.0-1062.4.3.el7.x86_64 #1 SMP Wed Nov 13 23:58:53 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux

```

## What's the configuration?

```
frontend acme.com_e63d09_frontend_1

bind unix@/run/haproxy/1_acme.com_0.sock user haproxy group nginx mode 0660
mode http
option httplog
option forwardfor except 127.0.0.1
option httpclose
default_backend acme.com_e63d09_backend_1

backend acme.com_e63d09_backend_1
balance leastconn
mode http
option httpclose
cookie SERVERID insert nocache indirect
timeout connect 50s
timeout server 50s

default-server alpn h2,http/1.1 agent-inter 3000 rise 2 fall 2 agent-check agent-port 10011 no-check agent-addr 127.0.0.1
server be_1_2_3_4 1.2.3.4:443 cookie srv0_4582_4 weight 100 agent-send eyJ0IjoidXJsIiwiaSI6IjEuMi4zLjQiLCJwIjo0NDMsIm0iOiJHRVQiLCJ1IjoiLyIsInIiOiJodHRwcyIsIngiOjMwMDB9\n
```

## Steps to reproduce the behavior

1. As per example configuration above, add the `agent-addr` directive to the default-server line. We are offloading checks to an external service, so all servers should get their checks from the same IP address.
2. Note that with `agent-addr` in `default-server`, the agent checks are sent to the backend servers rather than to localhost. However, the `agent-port` directive appears to have been obeyed as the checks are going to the backend server on port 10011.
3. Move the `agent-addr` directive to the server definition, note that the agent checks are now sent to the correct location (127.0.0.1:10011).

## Actual behavior

With `agent-addr` specified in `default-server`, it is ignored.

## Expected behavior

`agent-addr` in `default-server` should be honoured for all servers.

**Comment by wtarreau** [comment]
Hi, you're right it was indeed a bug since the inception of default-server. The problem is that the server's agent was always retrieved from the check's address and not the agent one, which is now fixed in 1.8 to 2.2-dev. However keep in mind that if you happen to set "addr" on your server line, it will stil overwrite both the check and the agent's addresses. It's not the case in your config though so you should be safe against this.

**Comment by Xiol** [comment]
In that case, could I request this is documented, please? I spent quite a bit of time debugging this and would like to save others the trouble!
